ReactNative/ReactTwitter/core/service/deep-linking-facade.ios.js

```diff
@@ -0,0 +1,30 @@
+import { Linking } from 'react-native'
+
+class DeepLinkingFacade {
+
+  static newInstance () {
+    return new DeepLinkingFacade(Linking)
+  }
+
+  constructor (linking) {
+    this.linking = linking
+  }
+
+  listenForDeepLinking () {
+    this.linking.removeEventListener('url', this._handleOpenURL)
+    this.linking.addEventListener('url', this._handleOpenURL)
+    return new Promise(function (resolve, reject) {
```

Comment:
No, I mean why are you removing and adding the event listener on every button click? So you can reset the promise?